packages/x-charts/src/LineChart/AreaPlot.tsx

```diff
@@ -81,9 +81,9 @@ const useAggregatedData = () => {
           const yScale = yAxis[yAxisId].scale;
           const xData = xAxis[xAxisId].data;
 
-          const gradientUsed: [AxisId, 'x' | 'y'] | undefined =
-            (yAxis[yAxisId].colorScale && [yAxisId, 'y']) ||
-            (xAxis[xAxisId].colorScale && [xAxisId, 'x']) ||
+          const gradientUsed: [AxisId] | undefined =
+            (yAxis[yAxisId].colorScale && [yAxisId]) ||
+            (xAxis[xAxisId].colorScale && [xAxisId]) ||
```

Comment:
This array was an easy way to pass typed arguments. now that the x/y is nto usefull, it could be removed. Either by just passing the axis id, or directly computing the gradient id here instead of doing `gradientId={gradientUsed && getGradientId(...gradientUsed)}` later

```suggestion
          const gradientUsed: AxisId | undefined =
            (yAxis[yAxisId].colorScale && yAxisId) ||
            (xAxis[xAxisId].colorScale && xAxisId) ||
```